designer/client/component-create.js

```diff
@@ -29,14 +36,15 @@ class ComponentCreate extends React.Component {
 
   render () {
     const { page, data } = this.props
+    const { id } = this.state
 
     return (
       <div>
         <form onSubmit={e => this.onSubmit(e)} autoComplete='off'>
           <div className='govuk-form-group'>
             <label className='govuk-label govuk-label--s' htmlFor='type'>Type</label>
             <select className='govuk-select' id='type' name='type' required
-              onChange={e => this.setState({ component: { type: e.target.value } })}>
+              onChange={async e => this.setState({ component: { type: e.target.value, name: id } })}>
```

Comment:
does this need to be async btw?

Reply:
Probably not, I was probably experimenting with something!

Reply:
Fixed